### PR TITLE
Optionally install Knative

### PR DESCRIPTION
**Pull request by kaxil**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -165,3 +165,13 @@ variable "enable_istio_local_gateway" {
   type    = bool
   default = false
 }
+
+variable "enable_knative" {
+  type    = bool
+  default = false
+}
+
```
Comment (sjmiller609) [inline variables.tf]: can we delete the variable enable_istio_local_gateway and use this variable instead? Also, can we please add a description to this variable to indicate that enable_istio must be true for this to work?

File: variables.tf [hunk 2]
```diff
@@ -165,3 +165,13 @@ variable "enable_istio_local_gateway" {
...
+variable "knative_helm_release_version" {
+  default     = "0.1.0"
```
Comment (sjmiller609) [inline variables.tf]: can we bump the chart version to match the knative version? I think it's 0.9.0

File: istio.tf [hunk 3]
```diff
@@ -52,3 +52,20 @@ resource "helm_release" "istio" {
...
 }
+
+resource "kubernetes_namespace" "knative_serving" {
+  count = var.enable_knative ? 1 : 0
+  metadata {
+    name = "knative-serving"
+  }
+}
+
+resource "helm_release" "knative" {
```
Comment (sjmiller609) [inline istio.tf]: can you please add helm_release.knative to the null_resource.dependency_setter depends_on in outputs.tf? this helps with cross-module dependencies.

Comment (kaxil) [reply istio.tf]: Done

File: variables.tf (hunk 1 above)
Comment (kaxil) [reply variables.tf]: Done

File: istio.tf [hunk 4]
```diff
@@ -50,5 +50,22 @@ resource "helm_release" "istio" {
   version    = var.istio_helm_release_version
   wait       = true
 
-  values = compact([var.enable_istio_local_gateway ? local.istio_local_gateway_helm_values : "", var.extra_istio_helm_values])
+  values = compact([var.knative_helm_release_version ? local.istio_local_gateway_helm_values : "", var.extra_istio_helm_values])
```
Comment (sjmiller609) [inline istio.tf]: I think this is a typo - should be var.enable_knativee

File: variables.tf (hunk 2 above)
Comment (kaxil) [reply variables.tf]: Not sure about this. I think we might be updating that chart based to fit our purpose so bumping to 0.9.1 then would not make sense.

Comment (sjmiller609) [reply variables.tf]: Ok that's fine, we can stick with this for now. I agree we don't want to change the helm chart without version bumping it.

File: istio.tf (hunk 4 above)
Comment (kaxil) [reply istio.tf]: You are right, thanks for spotting.

**Pull request merged**